build/scripts/build.ps1

```diff
@@ -636,6 +636,31 @@ function Stop-VSProcesses() {
     Get-Process -Name "devenv" -ErrorAction SilentlyContinue | Stop-Process
 }
 
+# Manually verify that the correct SDK is installed. This is necessary to guard 
+# against MSBuild silently downgrading to a different SDK.
+#
+# Issue tracking removing this check: https://github.com/dotnet/roslyn/issues/25378
+# Relevant issues:
+#   https://github.com/dotnet/announcements/issues/57
+#   https://github.com/dotnet/core-setup/issues/3805
+#   https://github.com/dotnet/core-setup/issues/3807
+function Check-BadDotnetSdkVersion() {
```

Comment:
Just to check: is this redundant with the MSBuild check as well? I don't mind the checking twice if that is the case, but I worry that those who build the solution directly with MSBuild will miss the check if they're _not_ redundant.

Reply:
This check should be rundandnt now.